tracer/src/Datadog.Trace/Configuration/ImmutableTracerSettings.cs

```diff
@@ -62,6 +63,10 @@ public ImmutableTracerSettings(TracerSettings settings)
             RouteTemplateResourceNamesEnabled = settings.RouteTemplateResourceNamesEnabled;
             DelayWcfInstrumentationEnabled = settings.DelayWcfInstrumentationEnabled;
 
+            // TODO: Should we load the API key in a normal way? We don't for AAS?
+            var apiKey = System.Environment.GetEnvironmentVariable(ConfigurationKeys.ApiKey);
```

Comment:
EnvironmentHelpers? 😄 
Also, I don't see this feature as tied to AAS.
I predict this to be used in all sorts of configurations, particularly when there are remote agents.
I would vote for the API key coming from normal configuration process.
I am guessing we'll get a future ticket about this if we only allow environment variables.

My only sticking point is to use our wrapper for getting the environment variable here though.

Reply:
agreed, and done in https://github.com/DataDog/dd-trace-dotnet/pull/2240/commits/52b41eb234fcd19500ffc132583cd2f6c9df8ec0